pkg/providers/vmware/client/client_test.go

```diff
@@ -0,0 +1,139 @@
+package client_test
+
+import (
+	"github.com/kubevirt/vm-import-operator/pkg/providers/vmware/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/simulator"
+)
+
+var _ = Describe("Test VMware rich client", func() {
+
+	DescribeTable("should connect to the host", func(model *simulator.Model) {
+		_ = model.Create()
+		server := model.Service.NewServer()
+		defer model.Remove()
+		defer server.Close()
+		username := server.URL.User.Username()
+		password, _ := server.URL.User.Password()
+
+		_, err := client.NewRichVMWareClient(server.URL.String(), username, password, "")
+		Expect(err).To(BeNil())
+	},
+		Entry("vCenter", simulator.VPX()),
+		Entry("ESXi", simulator.ESX()),
+	)
+
+	DescribeTable("should retrieve a VM by ID", func(model *simulator.Model) {
+		_ = model.Create()
+		server := model.Service.NewServer()
+		defer model.Remove()
+		defer server.Close()
+		vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+		username := server.URL.User.Username()
+		password, _ := server.URL.User.Password()
+		richClient, err := client.NewRichVMWareClient(server.URL.String(), username, password, "")
```

Comment:
The `richClient`creation code is the same in most of the test cases; it would be great if you found a way to extract as much of it as possible to a helper function.